### PR TITLE
Merge master to DIM before 3.0 switch

### PR DESCRIPTION
**Pull request by jaredpar**
Merges the master branch to features/DefaultInterfaceImplementation up to the 3.0 switch. That merge will be handled separately.

### CODE REVIEW

**Comment by jaredpar** [comment]
@AlekseyTs this is ready for review.

**Comment by jaredpar** [comment]
> The "Post merge cleanup" commit LGTM. Were there any other non-trivial changes worth looking at?

The other changes:

1. Dealing with some conflicts in the public API files for MS.CA and MS.CA.CSharp
1. Correcting some uses of `.TypeSymbol` to be `.Type`. This is just reacting to @gafter big rename PR. 

Looks like CI found one test error. Will dig into that.

File: eng/targets/Imports.targets [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
     <!-- Only generate our runtimeconfig.json files for net core apps. It's unnecessary in desktop projects
          but gets included in lots of output items like VSIX. -->
-    <GenerateRuntimeConfigurationFiles Condition="'$(TargetFramework)' != 'netcoreapp1.1' AND '$(TargetFramework)' != 'netcoreapp2.1'">false</GenerateRuntimeConfigurationFiles>
+    <GenerateRuntimeConfigurationFiles Condition="'$(TargetFramework)' != 'netcoreapp1.1' AND '$(TargetFramework)' != 'netcoreapp2.1' AND '$(TargetFramework)' != 'netcoreapp3.0'">false</GenerateRuntimeConfigurationFiles>
```
Comment (dsplaisted) [inline eng/targets/Imports.targets]: Ideally you should just have a condition based on `TargetFrameworkIdentifier`.  However, that only will work if this targets file is evaluated after `Microsoft.NET.TargetFrameworkInference.targets`, so I don't know if that's an option here.

Comment (jaredpar) [reply eng/targets/Imports.targets]: @dsplaisted this is hooked during Directory.Build.targets. Is that in the right place?

**Comment by jaredpar** [comment]
@AlekseyTs can you take a quick look at this commit which was necessary to fix a failing test? I took a look but can't tell if this is intended or not. It seems strange that this would have changed though. Worried there was a subtle merge issue https://github.com/dotnet/roslyn/pull/34263/commits/f5391a27341778ab246a29a06e9aac29c83d27a8

File: eng/targets/Imports.targets (hunk 1 above)
Comment (dsplaisted) [reply eng/targets/Imports.targets]: @jaredpar Yes, you should just be able to replace this condition with `'$(TargetFrameworkIdentifier)' != '.NETCoreApp'`.  On the other hand, as Nick mentioned in Slack, the whole condition is probably working around an old bug in the SDK and you can probably just delete the line entirely.

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceEventSymbol.cs [hunk 2]
```diff
@@ -533,14 +533,14 @@ protected void CheckModifiersAndType(DiagnosticBag diagnostics)
             {
                 // Diagnostic reported by parser.
             }
-            else if (!this.IsNoMoreVisibleThan(this.Type, ref useSiteDiagnostics) && (CSharpSyntaxNode as EventDeclarationSyntax)?.ExplicitInterfaceSpecifier == null)
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceEventSymbol.cs]: This change doesn't look right, I think this is what broke the ```ExplicitBase_149``` unit-test.

File: src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs [hunk 3]
```diff
@@ -55478,7 +55478,13 @@ internal interface C
             compilation0.VerifyDiagnostics(
                 // (2,18): error CS0061: Inconsistent accessibility: base interface 'C' is less accessible than interface 'B'
                 // public interface B : C
-                Diagnostic(ErrorCode.ERR_BadVisBaseInterface, "B").WithArguments("B", "C").WithLocation(2, 18)
+                Diagnostic(ErrorCode.ERR_BadVisBaseInterface, "B").WithArguments("B", "C").WithLocation(2, 18),
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs]: This points to a real issue.

**Comment by jaredpar** [comment]
@alekseyts fixed that still get the failure. Weird.

**Comment by AlekseyTs** [comment]
> fixed that still get the failure. Weird.

It looks like similar condition was dropped in src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

**Comment by jaredpar** [comment]
@Alekseyts given there were two merge errors I decided to re-do the merge. I'd lost confidence that I'd handled the conflicts correctly. Locally the symbols tests are passing now so feel better about this pass.

**Comment by jaredpar** [comment]
The debug 32 failure is a dupe of teh lange services finalization bug. All DIM and compiler test s passed.

**Pull request merged**